### PR TITLE
Update params

### LINKED ISSUE #251: bug: Constants in the model_mod for MPAS
I noticed that an atmospheric constant (specifically Cp or Specific heat of dry air at constant pressure) in model_mod.f90 in the mpas_atm directory does not match the value found in the MPAS source code.

DART Cp Constant Value (line 215 in model_mod.f90): 1003.0
MPAS Cp Constant Value (MPAS-Release/src/framework/mpas_constants.F: line 33): 1004.5

I believe you would want these constant to match up when converting potential temperature to temperature in the forward operator function (theta_to_tk line 7428).

### PR DESCRIPTION
**Pull request by syha**
## Description:
Parameters are updated in model_mod.f90 to be consistent with those in the MPAS-A model (Version 7+).

### Fixes issue

fixes #251
theta_m and pressure use rvord (=1.608362) instead of 1.61.
cp is now updated from 1003.0 to 1004.5 and cv is changed from 716.0 to 717.5.

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Documentation changes needed?
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### Tests
In 2-member tests with a single obs for sounding temperature at 700 hPa, new parameters changed all the analysis variables.

## Checklist for merging
- [ ] Updated changelog entry
- [ ] Documentation updated
- [ ] Update conf.py

## Checklist for release
- [ ] Merge into main
- [ ] Create release from the main branch with appropriate tag
- [ ] Delete feature-branch

## Testing Datasets
- [x] Dataset needed for testing available upon request
- [ ] Dataset download instructions included
- [ ] No dataset needed

### CODE REVIEW

File: models/mpas_atm/model_mod.f90 [hunk 1]
```diff
@@ -209,14 +209,15 @@ module model_mod
...
 ! Real (physical) constants as defined exactly in MPAS.
-! redefined here for consistency with the model.
-real(r8), parameter :: rgas = 287.0_r8
-real(r8), parameter :: rv = 461.6_r8
-real(r8), parameter :: cp = 1003.0_r8
-real(r8), parameter :: cv = 716.0_r8
+! redefined here for consistency with the model (MPAS/src/framework/mpas_constants.F).
+real(r8), parameter :: rgas = 287.0_r8  ! = R_d (Gas constant for dry air [J kg-1 K-1])
+real(r8), parameter :: rv = 461.6_r8    ! = R_v (Gas constant for water varpor [J kg-1 K-1])
+real(r8), parameter :: cp = 7.*rgas/2.  ! = 1004.5
```
Comment (hkershaw-brown) [inline models/mpas_atm/model_mod.f90:215]:
```suggestion
real(r8), parameter :: cp = 7.0_r8*rgas/2.0_r8  ! = 1004.5
```

**Comment by hkershaw-brown** [comment]
MPAS-A model (Version 5+) 
https://github.com/NCAR/DART/pull/429#issuecomment-1319351693

File: models/mpas_atm/model_mod.f90 [hunk 2]
```diff
@@ -7380,7 +7381,7 @@ function theta_to_tk (ens_size, theta, rho, qv, istatus)
 
 integer,                       intent(in)  :: ens_size
 real(r8), dimension(ens_size), intent(in)  :: theta    ! potential temperature [K]
-real(r8), dimension(ens_size), intent(in)  :: rho      ! dry density
+real(r8), dimension(ens_size), intent(in)  :: rho      ! dry air density [kg/m3]
```
Comment (hkershaw-brown) [inline models/mpas_atm/model_mod.f90:7384]: There are a bunch of subroutine arguments labelled as `dry density`. Should these all be `dry air density`?

Comment (hkershaw-brown) [reply models/mpas_atm/model_mod.f90:7384]: edit: it is one  in `compute_full_pressure`

File: models/mpas_atm/model_mod.f90 (hunk 1 above)
Comment (syha) [reply models/mpas_atm/model_mod.f90:215]: Looks better to me.

File: models/mpas_atm/model_mod.f90 (hunk 2 above)
Comment (syha) [reply models/mpas_atm/model_mod.f90:7384]: You are right. Here, rho is all 'dry air density'. Nice catch!

**Comment by kdraeder** [comment]
@soyoung closed #429 because the original pressure calculation appears to be correct,
but the commits to include a new qv^2 term seem to still be in this PR.
E.g. [Bug fix for full pressure with the additional term for qv^2.](https://github.com/NCAR/DART/pull/521/commits/ef4ac155fdadbc66d9ebbdd2333d8c548144fbfc)

File: models/mpas_atm/model_mod.f90 [hunk 3]
```diff
@@ -7457,7 +7458,7 @@ subroutine compute_full_pressure(ens_size, theta, rho, qv, pressure, tk, istatus
 tk = theta_to_tk(ens_size, theta, rho, qv_nonzero, istatus)
 
 where (istatus == 0)       ! We only take non-missing tk here
-   pressure = rho * rgas * tk * (1.0_r8 + 1.61_r8 * qv_nonzero)
+   pressure = rho * rgas * tk * (1.0_r8 + rvord * qv_nonzero)
```
Comment (nancycollins) [inline models/mpas_atm/model_mod.f90:7461]: i wanted to leave a comment on line 7445 - the 'dry density' comment should be 'dry air density' to be consistent.  i think that's the last place 'dry density' is in a comment (that i could find) that needs changing.  if all these suggested changes are ok, should we go ahead and accept them?

Comment (hkershaw-brown) [reply models/mpas_atm/model_mod.f90:7461]: @nancycollins I'm going to close this pull request and put the constant fix in cleanly. 
As Kevin says, it is probably best not to have the Jedi equations in the commits, and we need to add  documentation adding about the constant changes.

**Comment by hkershaw-brown** [comment]
closing - will redo without the Jedi commits.

File: models/mpas_atm/model_mod.f90 (hunk 3 above)
Comment (syha) [reply models/mpas_atm/model_mod.f90:7461]: @hkershaw-brown, just for clarification, there are no jedi equations or comments here at all. In this PR, I only updated the constants based on the model. I'm curious how you can make the fix cleaner?

Comment (hkershaw-brown) [reply models/mpas_atm/model_mod.f90:7461]: So the Jedi stuff is in the commit history:

Screen Shot 2023-08-02 at 4 31 25 PM

Comment (hkershaw-brown) [reply models/mpas_atm/model_mod.f90:7461]: @syha I'm aware you're busy with other things, so I'll put the fix in (I'll add you as a joint commit so you show up in the contributions). Thanks, Helen

Comment (syha) [reply models/mpas_atm/model_mod.f90:7461]: Ah, the commit history, not the code itself. Got it. Thanks!

**Pull request closed without merging**